modules/data.land/R/IC_BADM_Utilities.R

```diff
@@ -0,0 +1,296 @@
+
+#Reading the data and writing the netcdf
+#' Read.IC.info.BADM
+#'
+#' @param lat numeric latitude
+#' @param long numeric longitude
+#'
+#' @description This function returns a dataframe of plant biomass, root and soil carbon for a set of lat and long coordinates.
+#' This function first finds the level1 and level2 ecoregions  for the given coordinates, and then tries to filter BADM database for those eco-regions. 
+#' If no data found in the BADM database for the given lat/longs eco-regions, then all the data in the database will be used to return the initial condition.
+#' All the variables are also converted to kg/m^2. 
+#' @return a dataframe with 8 columns of Site, Variable, Date, Organ, PlantWIni (Initial plant biomass, type of biomass can be found in the Variable and Organ), SoilIni (which shows the initial soil C), LitterIni, RootIni.
+#'
+#' @export
+#' @examples
+#' \dontrun{
+#'   badm_test <- Read.IC.info.BADM(45.805925,-90.07961)
+#'}
+Read.IC.info.BADM <-function(lat, long){
+
+  #Reading in the DB
+  #
+  U.S.SB <-
+    readRDS(system.file("data","BADM.rds", package = "PEcAn.data.land"))
```

Comment:
Does it really make sense to store the entirety of BADM in Github rather than in BETY Inputs? Having it as a binary file in version control has the potential to rapidly bloat the repository as the contents of BADM evolves.

Reply:
I was afraid that the package/function may then become machine dependent. It's about 75 kb at moment. but If you think it's necessary I can register it in BETY as an input.

Reply:
If you keep it in the package, then I'd recommend storing it as text, not binary, so that any updates are stored as diffs, not as duplicate files. The fundamental problem with binary in version control is that every tiny change requires making a new copy of the file, and all copies of the file are forever stored and everyone who ever does a git clone has to download every version that's ever existed. This can be dangerous. With text, git can just store the incremental changes (additions/removals) which is much more efficient.